dts/bindings/i2s/nxp,mcux-i2s.yaml

```diff
@@ -64,6 +64,12 @@ properties:
       specifier-space: pinmux
       description: iomux settings
 
+    pinctrl-0:
+      required: true
+
+    pinctrl-names:
+      required: true
+
```

Comment:
Similar addition can be made to the uart `yaml` file

Reply:
We can't make an addition like this to to the uart `yaml` file until all boards supporting the uart driver also support pinctrl. If we add something like this, it effectively means all in tree boards with the `kinetis-lpuart` peripheral need to support pinctrl

This is because without defined pinctrl nodes for an LPUART node, adding these properties will cause the build to fail. It's the same issue that we have with using the `#ifdef CONFIG_PINCTRL` guards in the actual driver. In the case of the i2s driver, all boards that use it _must_ support pinctrl, so these properties are fine and the `ifdef` guards aren't needed.